web/template/index.html

```diff
@@ -66,7 +97,18 @@ <h2>Mainflux Deployment Summary</h2>
                 var toDate = toDateInput ? new Date(toDateInput).toISOString() : '';
 
                 var url = `/?from=${encodeURIComponent(fromDate)}&to=${encodeURIComponent(toDate)}`;
+                
+                window.location.href = url;
+            }
+
+            function applyCountryFilter() {
+                var selectedCountry = document.getElementById('country-filter').value;
+
+                var url = `/?country=${encodeURIComponent(selectedCountry)}`;
                 window.location.href = url;
+
+                // Prevent form submission (page reload)
+                return false;
             }
```

Comment:
move this to the apply filter method, all filters should be applied from the same form and have one callback to make the request